nengo/solvers.py

```diff
@@ -67,16 +72,14 @@ def mul_encoders(self, Y, E, copy=False):
         copy : bool, optional (Default: False)
             Whether a copy of ``Y`` should be returned if ``E`` is None.
         """
-        if self.weights:
-            if E is None:
-                raise ValidationError(
-                    "Encoders must be provided for weight solver", attr='E')
-            return np.dot(Y, E)
-        else:
-            if E is not None:
-                raise ValidationError(
-                    "Encoders must be 'None' for decoder solver", attr='E')
-            return Y.copy() if copy else Y
+        if self.weights and E is None:
+            raise ValidationError(
+                "Encoders must be provided for weight solver", attr='E')
+        if not self.weights and E is not None:
+            raise ValidationError(
+                "Encoders must be 'None' for decoder solver", attr='E')
+
+        return np.dot(Y, E) if E is not None else Y.copy() if copy else Y
```

Comment:
The nested conditionals here makes it a bit hard to parse... could you put `Y.copy() if copy else Y` in parentheses?